gem/lib/earlgrey/configure_earlgrey.rb

```diff
@@ -20,21 +20,33 @@ def configure_for_earlgrey(*args)
 
 module EarlGrey
   class << self
-    attr_reader :project_name, :test_target, :test_target_name, :scheme_file,
-                :user_project, :carthage, :swift, :swift_version
+    attr_reader :project_name, :installer, :test_target, :test_target_name, :scheme_file,
+              :user_project, :carthage, :swift, :swift_version
 
     # Returns path to Xcode file, prepending current working dir if necessary.
-    # @param xcode_file [String] xcode file path
+    # @param project_name [String] xcode file path
     # @param ext [String] xcode file extension
     # @return [String] path to Xcode file
-    def path_for(xcode_file, ext)
-      ext_match = File.extname(xcode_file) == ext
-      return xcode_file if File.exist?(xcode_file) && ext_match
-
-      path = File.join(Dir.pwd, File.basename(xcode_file, '.*') + ext)
+    def path_for(project_name, ext)
+      ext_match = File.extname(project_name) == ext
+      return project_name if File.exist?(project_name) && ext_match
+      path = File.join(dir_path, File.basename(project_name, '.*') + ext)
       path ? path : nil
     end
 
+    # Returns the project's directory. If CocoaPods hasn't had it passed in, then the current
+    # directory is chosen.
+    # @return [String] directory path for the Xcode project
+    def dir_path
+      if installer
+        pods_dir_path = installer.config.installation_root
```

Comment:
what is the value of `installer.config.installation_root` when `--project-directory` is not used?